variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/writers/VariantMongoWriterTest.java

```diff
@@ -0,0 +1,317 @@
+
+        variantMongoWriter.write(Collections.singletonList(variant1));
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        assertEquals(1, ((BasicDBList) storedVariant.get(FILES_FIELD)).size());
+    }
+
+    @Test
+    public void allFieldsOfVariantShouldBeStored() throws Exception {
+        final String chromosome = "12";
+        final long start = 3;
+        final long end = 4;
+        final String reference = "A";
+        final String alternate = "T";
+        final String fileId = "fileId";
+        final String studyId = "studyId";
+        Variant variant = buildVariant(chromosome, start, end, reference, alternate, fileId, studyId);
+
+        VariantMongoWriter variantMongoWriter = new VariantMongoWriter(collectionName, mongoOperations, false, true);
+        variantMongoWriter.write(Collections.singletonList(variant));
+
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        final BasicDBList variantSources = (BasicDBList) storedVariant.get(FILES_FIELD);
+        assertNotNull(variantSources);
+        assertFalse(variantSources.isEmpty());
+        assertEquals(fileId, ((DBObject) variantSources.get(0)).get(FILEID_FIELD));
+        assertEquals(studyId, ((DBObject) variantSources.get(0)).get(STUDYID_FIELD));
+        assertEquals(String.format("%s_%s_%s_%s", chromosome, start, reference, alternate), storedVariant.get("_id"));
+        assertEquals(chromosome, storedVariant.get(CHROMOSOME_FIELD));
+        assertEquals(start, storedVariant.get(START_FIELD));
+        assertEquals(end, storedVariant.get(END_FIELD));
+        assertEquals(reference, storedVariant.get(REFERENCE_FIELD));
+        assertEquals(alternate, storedVariant.get(ALTERNATE_FIELD));
+    }
+
+    @Test
+    public void includeStatsTrueShouldIncludeStatistics() throws Exception {
+        Variant variant = buildVariant("12", 3, 4, "A", "T", "fileId", "studyId");
+
+        VariantMongoWriter variantMongoWriter = new VariantMongoWriter(collectionName, mongoOperations, true, false);
+        variantMongoWriter.write(Collections.singletonList(variant));
+
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        assertNotNull(storedVariant.get(STATISTICS_FIELD));
+    }
+
+    @Test
+    public void includeStatsFalseShouldNotIncludeStatistics() throws Exception {
+        Variant variant = buildVariant("12", 3, 4, "A", "T", "fileId", "studyId");
+
+        VariantMongoWriter variantMongoWriter = new VariantMongoWriter(collectionName, mongoOperations, false, true);
```

Comment:
This test and the test before would be more interesting if just the "includeStats" parameter when building VariantMongoWriter is changed (now we are changing include samples too). If one of both is required, it should be made explicit in VariantMongoWriter (I don't see that check)